firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/serialization/MapEncoderKtxImp.kt

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.ktx.serialization
+
+import com.google.firebase.components.Component
+import com.google.firebase.firestore.encoding.MapEncoder
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.serializer
+
+class MapEncoderKtxImp : MapEncoder {
+    /**
+     * Encodes a custom serializable object to a nested map of Firebase firestore primitive types
+     */
+    override fun encode(value: Any): MutableMap<String, Any?> {
+        val serializer = serializer(value.javaClass)
+        return encodeToMap(serializer, value)
+    }
+
+    // a combination of isAbleToBeEncode and isAbleToBeDecoded
```

Comment:
Where are the two methods described in the doc?

Reply:
I have updated this Kotlin Doc to reflect that this one method servers both the encoding and decoding process.